### PR TITLE
refactor: Default device reformat in device.py

### LINKED ISSUE #26142: default_device

### PR DESCRIPTION
**Pull request by IamSolomonChika**
# PR Description
I reformatted the default_device function in ivy/functional/ivy/device.py
TASK FOR INTERNSHIP HIRING.

## Related Issue
None experienced.

Close #26142 

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### Socials
Twitter: @IamSolomonChika

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi, seems like you have changed `docs/demos` which you might have done by mistake. Can you revert that change because we can't merge with that change in place?

**Comment by IamSolomonChika** [comment]
@KareemMAX
I have synced my branch with the current update and the docs/demos file is no longer appearing to be changed.

File: ivy/functional/ivy/device.py [hunk 1]
```diff
@@ -818,12 +807,11 @@ def default_device(
...
-        if as_native is True:
+        if as_native:
             return ivy.as_native_dev(device)
-        elif as_native is False:
-            return ivy.as_ivy_dev(device)
-        return device
-    as_native = ivy.default(as_native, False)
+        return ivy.as_ivy_dev(device)
+    if not as_native:
+        as_native = False
```
Comment (KareemMAX) [inline ivy/functional/ivy/device.py:814]: This piece of code doesn't make sense. Can you elaborate on it?

File: ivy/functional/ivy/device.py [hunk 2]
```diff
@@ -767,36 +767,25 @@ def tpu_is_available() -> bool:
...
-    as_native: bool = None,
-) -> Union[ivy.Device, ivy.NativeDevice]:
+def default_device(device=None, *, item=None, as_native=None):
     """
     Return the input device or the default device. If the as_native flag is set, the
     device will be converted to a native device. If the item is provided, the item's
     device is returned. If the device is not provided, the last default device is
     returned. If a default device has not been set, the first gpu is returned if
-    available, otherwise the cpu is returned.
+    available, otherwise the cpu is returned..
```
Comment (KareemMAX) [inline ivy/functional/ivy/device.py:776]:
```suggestion
    available, otherwise the cpu is returned.
```

File: ivy/functional/ivy/device.py [hunk 3]
```diff
@@ -767,36 +767,25 @@ def tpu_is_available() -> bool:
...
 # noinspection PyShadowingNames
 @handle_exceptions
-def default_device(
-    device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
-    /,
-    *,
-    item: Optional[Union[list, tuple, dict, ivy.Array, ivy.NativeArray]] = None,
-    as_native: bool = None,
-) -> Union[ivy.Device, ivy.NativeDevice]:
+def default_device(device=None, *, item=None, as_native=None):
```
Comment (KareemMAX) [inline ivy/functional/ivy/device.py:770]: Why did you remove typehints?

File: ivy/functional/ivy/device.py [hunk 4]
```diff
@@ -767,36 +767,25 @@ def tpu_is_available() -> bool:
...
+    :param as_native: Whether to convert the device to a native device.
+    :type as_native: bool, optional
+    :return: Device handle or string.
+    :rtype: Union[ivy.Device, ivy.NativeDevice]
 
-    Returns
-    -------
-    ret
-        Device handle or string.
+    - **Example**
```
Comment (KareemMAX) [inline ivy/functional/ivy/device.py:787]: Why did you change the docstring style? We are using [numpy docstring style](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard).

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @IamSolomonChika,
I've pointed out some issues, as @Sam-Armstrong is your primary assignee, he will be taking over the review process.

File: ivy/functional/ivy/device.py (hunk 2 above)
Comment (IamSolomonChika) [reply ivy/functional/ivy/device.py:776]: this was a mistake, which I thought I removed

File: ivy/functional/ivy/device.py (hunk 4 above)
Comment (IamSolomonChika) [reply ivy/functional/ivy/device.py:787]: I will reverse this but I did it to clarify the hint that I removed in the function args

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
